client_test.go

```diff
@@ -173,3 +173,11 @@ func TestClient_RetrieveRecords_error(t *testing.T) {
 	_, err := client.RetrieveRecords(context.Background(), "example.com")
 	require.Error(t, err)
 }
+
+func TestClient_ErrorStatus(t *testing.T) {
+	client := setup(t, "/dns/retrieve/example.com", "error503")
+	_, err := client.RetrieveRecords(context.Background(), "example.com")
+	assert.IsType(t, &StatusError{}, err)
+	statusE := err.(*StatusError)
+	assert.Equal(t, "503", statusE.Status)
+}
```

Comment:
based on your curl, this test is not realistic.

Reply:
Could you please provide some guidance on this then? I'm not sure what is unrealistic or how I would improve it.

Reply:
Hi! Just wanted to bump this to see if you could provide some guidance about what is unrealistic? Thanks!